packages/jellyfish-api-core/__tests__/category/loan/getVault.test.ts

```diff
@@ -0,0 +1,191 @@
+      loanAmount: [new BigNumber(30).plus(new BigNumber(30).multipliedBy(interestInfo[0].totalInterest)).toFixed(8) + '@TSLA'], // 30.00000570@TSLA
+      // (10000 DFI * DFIUSD Price * DFI collaterization factor 1) + (1BTC * BTCUSD Price * BTC collaterization factor 0.5)
+      collateralValue: new BigNumber(10000 * 1 * 1).plus(new BigNumber(1 * 10000 * 0.5)),
+      // (30 TSLA + (30 TSLA * total interest)) * TSLAUSD Price
+      loanValue: new BigNumber(30).plus(new BigNumber(30).multipliedBy(interestInfo[0].totalInterest)).multipliedBy(2),
+      // lround ((collateral value / loan value) * 100)
+      currentRatio: new BigNumber(data.collateralValue?.dividedBy(data.loanValue as BigNumber).multipliedBy(100).toFixed(0, 4) as string)
+    })
+  })
+
+  it('should getVault with liquidated vault', async () => {
+    const ownerAddress = await testing.generateAddress()
+    const vaultId = await testing.rpc.container.call('createvault', [ownerAddress, 'default'])
+    await testing.generate(1)
+
+    await testing.container.call('deposittovault', [vaultId, collateralAddress, '10000@DFI'])
+    await testing.generate(1)
+    await testing.container.call('deposittovault', [vaultId, collateralAddress, '1@BTC'])
+    await testing.generate(1)
+
+    // take loan
+    await testing.container.call('takeloan', [{ vaultId: vaultId, amounts: '30@TSLA' }])
+    await testing.generate(1)
+
+    // check vault not under liquidation.
+    const data = await testing.rpc.loan.getVault(vaultId)
+    expect(data.isUnderLiquidation).toStrictEqual(false)
+
+    // make vault enter under liquidation state by a price hike of the loan token
+    const timestamp = Math.floor(new Date().getTime() / 1000)
+    await testing.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '1000@TSLA', currency: 'USD' }] })
+    await testing.generate(1)
+
+    // get auction details
+    const autionDetails: [] = await testing.rpc.call('listauctions', [], 'bignumber')
+
+    const vaultDataAfterPriceHike = await testing.rpc.loan.getVault(vaultId)
+    expect(vaultDataAfterPriceHike).toStrictEqual({
+      loanSchemeId: 'default', // Get default loan scheme
+      ownerAddress: ownerAddress,
+      isUnderLiquidation: true,
+      batches: autionDetails.filter((auction: {vaultId: string}) => auction.vaultId === vaultId).map((auction: {batches: []}) => auction.batches)[0]
+    })
+
+    // set the price oracle back to original price
+    await testing.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '2@TSLA', currency: 'USD' }] })
+  })
+
+  it('should not getVault if vault id is invalid', async () => {
+    // Pass non existing hex id
+    const promise = testing.rpc.loan.getVault('2cca2e3be0504af2daac12255cb5a691447e0aa3c9ca9120fb634a96010d2b4f')
+    await expect(promise).rejects.toThrow('RpcApiError: \'Vault <2cca2e3be0504af2daac12255cb5a691447e0aa3c9ca9120fb634a96010d2b4f> not found\', code: -20, method: getvault')
```

Comment:
promise1, promise2, promise3

Reply:
Or even better, adding open and close braces, so you can reuse the same variable `promise`